pretyping/nativenorm.ml

```diff
@@ -149,30 +139,31 @@ let construct_of_constr_const env sigma tag typ =
 
 let construct_of_constr_block = construct_of_constr false
 
-let build_branches_type env sigma (mind,_ as _ind) mib mip u params p =
+let build_branches_type env sigma mib mip indu params p =
   let rtbl = mip.mind_reloc_tbl in
+  let paramsl = Array.to_list params in
   (* [build_one_branch i cty] construit le type de la ieme branche (commence
      a 0) et les lambda correspondant aux realargs *)
-  let build_one_branch i cty =
-    let typi = type_constructor mind mib u cty params in
+  let build_one_branch i =
+    let typi = Inductiveops.instantiate_constructor_params indu (mib,mip) i paramsl in
     let decl,indapp = Reductionops.splay_prod env sigma (EConstr.of_constr typi) in
     let decl = List.map (on_snd EConstr.Unsafe.to_constr) decl in
     let decl_with_letin,_ = decompose_prod_assum typi in
     let ind,cargs = find_rectype_a env sigma indapp in
     let nparams = Array.length params in
-    let carity = snd (rtbl.(i)) in
+    let carity = snd (rtbl.(i-1)) in
     let crealargs = Array.sub cargs nparams (Array.length cargs - nparams) in
     let codom =
       let ndecl = List.length decl in
       let papp = mkApp(lift ndecl p,crealargs) in
-      let cstr = ith_constructor_of_inductive (fst ind) (i+1) in
+      let cstr = ith_constructor_of_inductive (fst ind) i in
       let relargs = Array.init carity (fun i -> mkRel (carity-i)) in
       let params = Array.map (lift ndecl) params in
       let dep_cstr = mkApp(mkApp(mkConstructU (cstr,snd ind),params),relargs) in
       mkApp(papp,[|dep_cstr|])
     in
     decl, decl_with_letin, codom
-  in Array.mapi build_one_branch mip.mind_nf_lc
+  in Array.init 1 build_one_branch
```

Comment:
Why does the returned array always have one single cell? If the code is actually correct, it would be better not to return an array in the first place.

Reply:
It is a stupid confusion on my side (as CI says). Should be `Array.length mip.mind_nf_lc`. Fixed and test-suite now happy.